rs/tests/qualification/qualification.rs

```diff
@@ -0,0 +1,182 @@
+use ic_protobuf::registry::subnet::v1::SubnetType;
+use ic_system_test_driver::driver::{
+    group::SystemTestGroup,
+    test_env_api::{read_dependency_from_env_to_string, read_dependency_to_string},
+};
+use ic_tests::qualification::{
+    defs::QualificationExecutor,
+    steps::{
+        ensure_blessed_version::EnsureBlessedVersion,
+        retire_blessed_version::RetireBlessedVersions, update_subnet_type::UpdateSubnetType,
+        workload::Workload, xnet::XNet,
+    },
+    ConfigurableSubnet, ConfigurableUnassignedNodes, IcConfig, SubnetSimple,
+};
+use std::time::Duration;
+
+// 4 Hours
+const OVERALL_TIMEOUT: Duration = Duration::from_secs(4 * 60 * 60);
+
+pub fn main() -> anyhow::Result<()> {
+    // setup env variable for config
+    let old_version = match std::env::var("OLD_VERSION") {
+        Ok(v) => v,
+        Err(_) => read_dependency_to_string("testnet/mainnet_nns_revision.txt").map_err(|_| anyhow::anyhow!("Didn't find initial version specified in `testnet/mainnet_nns_revision.txt` nur in `INITIAL_VERSION` env variable"))?,
+    };
+    let new_version = match std::env::var("NEW_VERSION") {
+        Ok(v) => Some(v),
+        Err(_) => None,
+    };
+
+    let config = IcConfig {
+        subnets: Some(vec![
+            ConfigurableSubnet::Simple(SubnetSimple {
+                subnet_type: ic_registry_subnet_type::SubnetType::System,
+                num_nodes: 4,
+            }),
+            ConfigurableSubnet::Simple(SubnetSimple {
+                subnet_type: ic_registry_subnet_type::SubnetType::Application,
+                num_nodes: 4,
+            }),
+            ConfigurableSubnet::Simple(SubnetSimple {
+                subnet_type: ic_registry_subnet_type::SubnetType::Application,
+                num_nodes: 4,
+            }),
+        ]),
+        unassigned_nodes: Some(ConfigurableUnassignedNodes::Simple(4)),
+        boundary_nodes: None,
+        // If both versions are specified its safe to start from
+        // the old version. If we didn't specify the new version
+        // it means that we are running from the tip of the branch
+        // and images will not be present.
+        initial_version: new_version.as_ref().map(|_| old_version.clone()),
```

Comment:
Maybe this is more readable
```suggestion
        initial_version: new_version.is_some().then_some(old_version.clone()),
```